### DISCUSSION TITLE
Access to Selective Private Series and Chapters to certain Users

### DISCUSSION THREAD

**Discussion by bravking** [Tetrakern/fictioneer#41] [Ideas]
Is it possible to extend the assign unlock posts for private series and chapters (posts)? I'm currently publishing beta series privately, but I'm facing a challenge with access control. I could create a custom role 'Beta Readers' with "read private posts" capability but the standard "read private posts" capability is too broad, as it grants access to all private content. What I need is a way to assign specific private series to particular users or groups, rather than giving blanket access to all private posts. Is there a feasible way to implement this kind of granular access control?

**Comment by Tetrakern** [comment]
Feasible, yes. Reasonable? I'll have to think about that.

However, you can also use the Fictioneer Role Manager and use the special "Read Private Chapters/Stories" capabilities. They should do what you want and not grant access to all private content. Well, certainly all private chapters and stories, though.

**Reply by dreaktr** [reply]
I understand what the OP is looking for, as I also provide access to pre-released content for private readers. Currently, I share Google Docs links for these chapters, but I'll admit it's not the most reader-friendly solution.

I believe one way you could do it is by expanding the existing protected-stories feature in Fictioneer. However, I'm wondering about the practicalities of implementation, particularly regarding user permissions.

The main challenge I see is allowing specific users to view individual private stories and chapters - similar to admin access - without granting them full administrative privileges.

So, I'm wondering if you, Tetrakern, given your expertise, do you have any insights on how this feature extension might work within the WP framework? Are there potential solutions or limitations we should be aware of? I'm curious to hear your thoughts on this.

**Reply by Tetrakern** [reply]
Unless I'm completely off topic, you can already do that. Fictioneer's Unlock feature is built around protected posts (password), which is best for compatibility reasons (caching, etc.). Private and scheduled posts are normally not accessible without specific capabilities, which opens another can of worms.

For instance, you could just roll your head over your keyboard to generate a password that is never shared and use the Unlock feature to allow logged-in users to bypass said password. Or you set up a role that allows that for a whole post type.

However, you can also allow user roles (Fictioneer > Roles) to access private posts, stories, chapter, etc. **separately** regardless. You cannot do that with scheduled content, but that is not supposed to be visible or accessible by design. If you change that, you are on your own.

You can also unlock posts/etc. directly for a user on the user's admin profile page. Not particularly convenient if you got many users, but it's doable.

<img width="874" alt="image" src="https://github.com/user-attachments/assets/981b65bf-ff5e-4772-8fba-8e2e778b39ff">

The only thing missing is that you cannot unlock posts/etc. for user roles. You can for Patreon tiers, though. See, this feature was sponsored by someone and built to their requirements. And they wanted this for Patreon.
